### PR TITLE
Use transactions in more `pgdb` functions

### PR DESCRIPTION
**Pull request by AlekSi**
## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by w84thesun — APPROVED** [review]

**Review by noisersup — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

File: internal/handlers/pg/pgdb/collections.go [hunk 1]
```diff
@@ -196,10 +190,10 @@ func CreateCollectionIfNotExist(ctx context.Context, querier pgxtype.Querier, db
 
 // DropCollection drops FerretDB collection.
 //
-// It returns (possibly wrapped) ErrTableNotExist if schema or table does not exist.
+// It returns (possibly wrapped) ErrTableNotExist if database or collection does not exist.
```
Comment (rumyantseva) [inline internal/handlers/pg/pgdb/collections.go:193]: There is some confusion here. The function is called `DropCollection` and the comment contains `database/colletion` things, but the error is called `ErrTableNotExist`... It's probably ok, but not 100% consistent.

**Review by rumyantseva — APPROVED** [review]
That looks like a helpful improvement!

File: internal/handlers/pg/pgdb/collections.go (hunk 1 above)
Comment (AlekSi) [reply internal/handlers/pg/pgdb/collections.go:193]: Yeah, we will fix it later. One step at a time

Comment (rumyantseva) [reply internal/handlers/pg/pgdb/collections.go:193]: Makes sense!

**Pull request merged**